packages/react-form/src/hooks/form.ts

```diff
@@ -61,21 +62,34 @@ export function useForm<T extends FieldBag>({
   onSubmit?: SubmitHandler<FormMapping<T, 'value'>>;
 }): Form<T> {
   const dirty = useDirty(fields);
-  const reset = useReset(fields);
+  const basicReset = useReset(fields);
   const {submit, submitting, errors, setErrors} = useSubmit(onSubmit, fields);
 
+  const reset = useCallback(
+    () => {
+      setErrors([]);
+      basicReset();
+    },
+    [basicReset, setErrors],
+  );
+
+  const fieldsRef = useRef(fields);
+  fieldsRef.current = fields;
+
+  const validate = useCallback(
+    () => {
+      return validateAll(fieldsRef.current);
```

Comment:
I suppose we could bring back `validateAll`  as a discrete hook now possibly, but I'm not sure that actually is much better.